File: tests/fixtures/edgecase_repo/scala/Service5.scala
package com.example.gen

import com.example.gen.Base5

trait Handler5 {
  def handle(payload: String): Boolean
}

class Base5(val name: String)

class Service5(name: String) extends Base5(name) with Handler5 {
  def handle(payload: String): Boolean = {
    this.validate(payload)
    true
  }

  private def validate(payload: String): Unit = {}
}

object Helpers5 {
  def helper(value: String): String = value
}

File: tests/fixtures/edgecase_repo/swift/Service5.swift
import Foundation

protocol Handler5 {
    func handle(_ payload: String) -> Bool
}

class Base5 {
    let name: String
    init(name: String) {
        self.name = name
    }
}

class Service5: Base5, Handler5 {
    func handle(_ payload: String) -> Bool {
        self.validate(payload)
        return true
    }

    private func validate(_ payload: String) {}
}

func helper5(_ value: String) -> String {
    return value
}
